concordium-consensus/src/Concordium/GlobalState/BlockState.hs

```diff
@@ -1530,6 +1530,19 @@ class (BlockStateQuery m) => BlockStateOperations m where
     --  round did timeout.
     bsoUpdateMissedRounds :: (PVSupportsDelegation (MPV m), PVSupportsValidatorSuspension (MPV m)) => UpdatableBlockState m -> Map.Map BakerId Word64 -> m (UpdatableBlockState m)
 
+    -- | Mark given validators for possible suspension at the next snapshot
+    --  epoch. Returns the subset of the given validator ids whose missed rounds
+    --  exceeded the given threshold and are now primed for suspension.
+    bsoPrimeForSuspension :: (PVSupportsDelegation (MPV m), PVSupportsValidatorSuspension (MPV m)) => UpdatableBlockState m -> Word64 -> [BakerId] -> m ([BakerId], UpdatableBlockState m)
```

Comment:
Added some more docs for the parameters.

Reply:
I think it's better to pass both parameters explicitly. Because this function is meant to be called with the current paydays validators. If they are not passed in explicitly, we would have to gather them (most likely from the `BakerPoolRewardDetails`) during the call. The same holds for the threshold, we could also gather it by inspecting the chain parameters. But I think being explicit makes the intent a lot clearer and doesn't rely on implicit assumptions on the implementations.